watertap/examples/flowsheets/case_studies/full_water_resource_recovery_facility/BSM2_P_extension.py

```diff
@@ -665,7 +764,45 @@ def initialize_system(m, bio_P=False):
     seq.set_guesses_for(m.fs.translator_asm2d_adm1.inlet, tear_guesses2)
 
     def function(unit):
-        unit.initialize(outlvl=idaeslog.INFO)
+        # unit.initialize(outlvl=idaeslog.INFO)
+        if unit == m.fs.translator_asm2d_adm1:
+            try:
+                print("Trying to initialize ASM2d-ADM1 translator")
+                unit.initialize(outlvl=idaeslog.DEBUG)
+            except:
+                print("Entering exception clause")
+                m.fs.translator_asm2d_adm1.inlet.flow_vol.fix()
+                m.fs.translator_asm2d_adm1.inlet.conc_mass_comp.fix()
+                m.fs.translator_asm2d_adm1.inlet.temperature.fix()
+                m.fs.translator_asm2d_adm1.inlet.pressure.fix()
+
+                solver = pyo.SolverFactory("ipopt")
+                solver.solve(m.fs.translator_asm2d_adm1, tee=True)
+
```

Comment:
Do you add this because the solve doesn't work without this? Or is this due to transcribing improvements from Soraya? Just want to point out that the default solver we use is ipopt-watertap which is basically a tweaked version of ipopt that ends up being more suitable for problems in watertap. However, by calling ipopt here, you are switching to the IDAES version of ipopt, without the applied tweaks meant for solving watertap models. If you simply delete line 779, you'll see if this solves with the default ipopt-watertap

Reply:
I was mainly using this to try to get an initial solution that would help with providing better tear guesses. This is no longer necessary, so it's been deleted. Thanks for the ipopt clarification too!